packages/extension-seo-url-mapping/src/setup-app.tsx

```diff
@@ -0,0 +1,110 @@
+} from '@salesforce/pwa-kit-extension-sdk/react'
+import {applyHOCs} from '@salesforce/pwa-kit-extension-sdk/react/utils'
+import {transformUrlMappingToRoute} from '@salesforce/pwa-kit-react-sdk/utils/routes'
+
+// Local Imports
+import {Config} from './types'
+import {SerializedRoute} from './types/routes'
+import {getUrlMapping} from './utils/routes'
+
+// Others
+import extensionMeta from '../extension-meta.json'
+
+interface StoreSlice {
+    count: number
+    increment: () => void
+    decrement: () => void
+}
+
+// This is safe to delete if your extension does not use state. If you aren't using this, ensure you remove the
+// `withApplicationExtensionStore` usage below as well.
+const sliceInitializer: SliceInitializer<StoreSlice> = (set) => ({
+    count: 0,
+    increment: () => set((state) => ({count: state.count + 1})),
+    decrement: () => set((state) => ({count: state.count - 1}))
+})
+
+// TODO: change name of extension
+class SeoUrlMappingExtension extends ApplicationExtension<Config> {
+    static readonly id = extensionMeta.id
+
+    /**
+     * Use this method to wrap or enhance your PWA-Kit application using [React higher-order components](https://legacy.reactjs.org/docs/higher-order-components.html).
+     * You can use this to add visual treatments to your application, change the props that are supplied to the application component
+     * or add things like providers and contexts to be used throughout your app.
+     */
+    extendApp<T extends React.ComponentType<T>>(
+        App: React.ComponentType<T>
+    ): React.ComponentType<T> {
+        const HOCs = [
+            // Optionally include state for this extension using `withApplicationExtensionStore`
+            (component: React.ComponentType<any>) =>
+                withApplicationExtensionStore(component, {
+                    id: extensionMeta.id,
+                    initializer: sliceInitializer
+                })
+        ]
+
+        return applyHOCs(App, HOCs)
+    }
+
+    /**
+     * This method is used to make changes to the PWA-Kit application routes. If your extension adds a new page to the application
+     * then you can add it to the router here. The routes passed to this method is an accrued list of routes that have been added
+     * from extensions applied before it. It is called during the `getRoutes` phase on both the server and the client.
+     *
+     * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
+     */
+    async extendRoutes(routes: RouteProps[], req?: Request): Promise<RouteProps[]> {
+        const path = req?.path
+        const {resourceTypeToComponentMap} = this.getConfig()
+        // TODO: How to call urlMapping API when React Hook "useUrlMapping"
+        // cannot be called in a class component. React Hooks must be called
+        // in a React function component or a custom React Hook function.
+        const mapping = await getUrlMapping(path)
```

Comment:
I think you briefly mentioned this but the `useUrlMapping` hook from `commerce-sdk-react` can't be called here because it needs to be used within a React hook or component. 

For now instead of making an actual API call I made a mock function `getUrlMapping` that returns the response for the following types of urlMappings: `product`, `category`, `redirect`, `content_asset`

Reply:
Please reach out the extensibility channel so we can start brain storming some idea's to solve this problem. I think this is potentially a blocker to getting this implemented.